eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -1170,32 +1170,46 @@ registerCommitteeHotKey ::
   Credential 'ColdCommitteeRole (EraCrypto era) ->
   ImpTestM era (Credential 'HotCommitteeRole (EraCrypto era))
 registerCommitteeHotKey coldKey = do
-  hotKey NE.:| [] <- registerCommitteeHotKeys $ pure coldKey
+  hotKey NE.:| [] <- registerCommitteeHotKeys (KeyHashObj <$> freshKeyHash) $ pure coldKey
   pure hotKey
 
 registerCommitteeHotKeys ::
   (ShelleyEraImp era, ConwayEraTxCert era) =>
+  -- | Hot Credential generator
+  ImpTestM era (Credential 'HotCommitteeRole (EraCrypto era)) ->
   NonEmpty (Credential 'ColdCommitteeRole (EraCrypto era)) ->
   ImpTestM era (NonEmpty (Credential 'HotCommitteeRole (EraCrypto era)))
-registerCommitteeHotKeys coldKeys = do
-  keys <- forM coldKeys (\coldKey -> (,) coldKey . KeyHashObj <$> freshKeyHash)
+registerCommitteeHotKeys genHotCred coldKeys = do
```

Comment:
This is so that we can also pass it a particular hot key as parameter right?

Reply:
Correct. This way we have control of which hot key we use. For example this allowed me to authorize the same hot key for all cold credentials